petric.py

```diff
@@ -170,9 +171,18 @@ def construct_RDP(penalty_strength, initial_image, kappa, max_scaling=1e-3):
     return prior
 
 
-Dataset = namedtuple('Dataset', [
-    'acquired_data', 'additive_term', 'mult_factors', 'OSEM_image', 'prior', 'kappa', 'reference_image',
-    'whole_object_mask', 'background_mask', 'voi_masks'])
+@dataclass
+class Dataset:
+    acquired_data: STIR.AcquisitionData
+    additive_term: STIR.AcquisitionData
+    mult_factors: STIR.AcquisitionData
+    OSEM_image: STIR.ImageData
+    prior: STIR.RelativeDifferencePrior | Any
```

Comment:
I guess `Any` to be able to cope with `CudaRelativeDifferencePrior`? I'm not so familiar with Python typing, but note that `CudaRelativeDifferencePrior` is derived from `RelativeDifferencePrior` so maybe we don't need `Any`?

Reply:
yup

```suggestion
    prior: STIR.RelativeDifferencePrior
```